### ISSUE TITLE
Add media-feature-name-*list rules

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#1722]
- [ ] `media-feature-name-blacklist`
- [ ] `media-feature-name-whitelist`

This proposal comes via a [Stack Overflow support request](http://stackoverflow.com/questions/38572426/how-to-use-stylelint-to-prevent-use-of-max-width-in-media-queries) that I found no other way to solve.

Two new rules `media-feature-blacklist` & `media-feature-whitelist`

Specify a blacklist/whitelist of disallowed/allowed media features:

```css
@media (max-width: 50em) { padding: 1rem; }
/**     ↑
 *       Media features like this */
```

Options

`array|string`: `["array", "of", "unprefixed", "media-features"]|"media-feature"`

Given:

```js
  "media-feature-blacklist": [
    "max-width",
  ],
  "media-feature-whitelist": [
    "min-width",
  ],
```

The following patterns are considered warnings:

```css
@media (max-width: 50em) {
  padding: 1rem;
}
```

The following patterns are *not* considered warnings:
```css
@media (min-width: 50em) {
  padding: 1rem;
}
```

----

Reference: MDN [Media Features](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features)

**Comment by evilebottnawi** [comment]
sgtm

**Comment by jeddy3** [comment]
SGTM. Writing this proposal up was on my list of things todo this morning :)

**Comment by ntwb** [comment]
Related:  #1711 introduces `mediaFeatureNames` and `deprecatedMediaFeatureNames` keyword sets

**Comment by jeddy3** [comment]
- Names:
  - `media-feature-name-blacklist`
  - `media-feature-name-whitelist`
- Primary options: `["string", "/regex/"]`
- Secondary options: none
- Messages: Unexpected media feature name "${name}"`
- Section: "Media feature"

**Comment by jeddy3** [comment]
@aaronbushnell Would you consider adding these rules? 

I recently created the equivalent rules for pseudo classes, as such there are solid PR blueprints for each rule in https://github.com/stylelint/stylelint/pull/1695 and https://github.com/stylelint/stylelint/pull/1667. Instead of using the selector parser, like I did, you'll just need to swap it out for the media parser. There's [an example](https://github.com/stylelint/stylelint/blob/master/src/rules/media-feature-name-case/index.js#L27-L29) of how that parser works in [`media-feature-name-case`](https://github.com/stylelint/stylelint/blob/master/src/rules/media-feature-name-case/).

**Comment by evilebottnawi** [comment]
@ntwb good idea, use `media-feature-blacklist` to disable use `deprecated` `media feature name`

**Comment by woble** [comment]
This wouldn't work if media features are writting in a range context, e.g. `@media (width <= 50em)`, unless stylelint transpiles and then checks. But that could be confusing if what is speficied in the configuration does not correspond with how CSS is written.

**Comment by jeddy3** [comment]
@woble As per #1728, we'll ignore media features in a range context for now. Adding support in `future-major`. I don't see any incompatibility in the rule signature itself, we just currently lack the means to reliably extract the media feature names from media features in a range context.

> Edit: Perhaps a separate black/white list rule can be created to handle comparison operators?

Feel free to create a new issue to discuss that.

**Comment by aaronbushnell** [comment]
Sorry all, I'm unfortunately too tied up to dig into this more and get something going. I'm hoping I can get to this eventually.

I did just wanted to say I appreciate everyone being really proactive and taking a Stack Overflow question all the way up here! Thanks for doing all that! 😃

**Comment by gaidarenko** [comment]
Please label this issue as `wip`.

**Comment by evilebottnawi** [comment]
@gaidarenko let's rock

**Comment by ntwb** [comment]
Thanks @gaidarenko 🎉
